source/extensions/filters/network/zookeeper_proxy/filter.cc

```diff
@@ -288,24 +325,29 @@ void ZooKeeperFilter::onCloseRequest() {
   setDynamicMetadata("opname", "close");
 }
 
-void ZooKeeperFilter::onConnectResponse(const int32_t proto_version, const int32_t timeout,
-                                        const bool readonly,
+void ZooKeeperFilter::onConnectResponse(const OpCodes opcode, const int32_t proto_version,
+                                        const int32_t timeout, const bool readonly,
                                         const std::chrono::milliseconds& latency) {
-  config_->stats_.connect_resp_.inc();
+  std::string opname = onResponseHelper(opcode, latency);
 
-  Stats::Histogram& histogram = Stats::Utility::histogramFromElements(
-      config_->scope_, {config_->stat_prefix_, config_->connect_latency_},
-      Stats::Histogram::Unit::Milliseconds);
-  histogram.recordValue(latency.count());
-
-  setDynamicMetadata({{"opname", "connect_response"},
+  setDynamicMetadata({{"opname", opname},
                       {"protocol_version", std::to_string(proto_version)},
                       {"timeout", std::to_string(timeout)},
                       {"readonly", std::to_string(readonly)}});
 }
 
 void ZooKeeperFilter::onResponse(const OpCodes opcode, const int32_t xid, const int64_t zxid,
                                  const int32_t error, const std::chrono::milliseconds& latency) {
+  std::string opname = onResponseHelper(opcode, latency);
+
+  setDynamicMetadata({{"opname", opname},
+                      {"xid", std::to_string(xid)},
+                      {"zxid", std::to_string(zxid)},
+                      {"error", std::to_string(error)}});
+}
+
+std::string ZooKeeperFilter::onResponseHelper(const OpCodes opcode,
```

Comment:
it's vague to me what `onResponseHelper` does and returns.
this might be clear but i'll let you judge
```
onResponse {
opinfo = getOpinfo(opcode);
std::string = opname{};
if (opinfo != unknown) {
  opname = opinfo.name_;
  recordErrorBudget();
}
setDynamicData(...)
}
```